pkg/shell/cockpit-dashboard.js

```diff
@@ -410,7 +410,24 @@ PageDashboard.prototype = {
                 update_series();
             }
 
-            return render;
+            function throttled_render() {
+                var timer = null;
+                var needs_render = false;
+                return function () {
+                    if (timer) {
+                        needs_render = true;
+                    } else {
+                        render();
+                        timer = window.setTimeout(function () {
+                            if (needs_render)
+                                render();
```

Comment:
For a delay (as you clarified in your comment) you can just always render here and skip the first render call above. This would also mean fewer unnecessary render calls.